File: android/app/src/main/java/dev/deliteai/assistant/domain/features/llm/LLMService.kt
/*
 * SPDX-FileCopyrightText: (C) 2025 DeliteAI Authors
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package ai.nimbleedge.nimbleedge_chatbot.domain.features.llm

import ai.nimbleedge.datamodels.NimbleNetTensor

object LLMService {
    suspend fun feedInput(input:String, isVoiceInitiated:Boolean) = LLMManager.feedInput(input, isVoiceInitiated)
    suspend fun getNextMap(): Map<String, NimbleNetTensor> = LLMManager.getNextMap()
    suspend fun stopLLM() = LLMManager.stopLLM()
    suspend fun getLLMName() = LLMManager.getLLMName()
}

File: ios/NimbleEdgeAI/feature/llm/LLMService.swift
/*
 * SPDX-FileCopyrightText: (C) 2025 DeliteAI Authors
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import Foundation
import DeliteAI
class LLMService {
    func feedInput(input: String, isVoiceInitiated: Bool = false) async throws {
        try await LLMManager.feedInput(input: input, isVoiceInitiated: isVoiceInitiated)
    }

    func getNextMap() async throws ->  [String: NimbleNetTensor] {
        try await LLMManager.getNextMap()
    }
    func stopLLM ()  throws{
        try LLMManager.stopLLM()
    }
}
